docs/content/basics/testing_theory.md

```diff
@@ -1,57 +1,163 @@
+   }
+}
+```
+
+In order to write a shared test, it is important to use `androidx.test.ext.junit.runners.AndroidJunit4` by annotating them with ``@RunWith(AndroidJUnit4::class)``. Why? Take a look at the implementation of AndroidJunit4. The magic actually happens in this method
+
+```kotlin
+private static String getRunnerClassName() {
+    String runnerClassName = System.getProperty("android.junit.runner", null);
+    if (runnerClassName == null) {
+      if (!System.getProperty("java.runtime.name").toLowerCase().contains("android")
+          && hasClass("org.robolectric.RobolectricTestRunner")) {
+        return "org.robolectric.RobolectricTestRunner";
+      } else {
+        return "androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner";
+      }
+    }
+    return runnerClassName;
+}
+```
+The `System.getProperty("android.junit.runner", null)` acts as a switch to detect the environment where the test is running and decide whether to use the `RobolectricTestRunner` or the `AndroidJUnit4ClassRunner` accordingly.
+
+Once you have your test written in the sharedTestFolder, it is ready to run on any environment! 
+In order to run it on the JVM, execute a command similar with the following structure
+
+```
+./gradlew :MODULE:testVARIANTUnitTest --tests "PACKAGE.CLASS"
+```
+
+and to run it on the device/emulator, the command would have this format
+```
+./gradlew :MODULE:connectedVARIANTAndroidTest -Pandroid.testInstrumentationRunnerArguments.class=PACKAGE.CLASS
+```
+
+#### Benefits
+Ui tests running on the JVM are much faster than those running on a device/emulator. On the other hand, we lose reliability: these tests are running with Robolectric, which mocks most parts of the Android system.
+That means that, even though a tests passes on the JVM, it might not on a device, in the real environment where our app runs.
+By writing tests that might run on the JVM or on a real device, we can get the best of both worlds:
+
+1. Run them on the JVM on every PR. Since they are fast, our builds times won't increase too much while we'll get feedback if sth does not work as expected
+2. Run them on a device/emulator once a day/before release. Since they are much slower, build times will be long, but we are ensuring the tests pass when running on a real environment.
+
+
+#### Limitations
+Keep in mind that such tests have the same limitations as Robolectric
+
+1. Do not support Idling resources yet. [See issue here](https://github.com/robolectric/robolectric/issues/4807)
+2. Do not support (and will not) tests that start new activities (i.e. activity jumping). [See issue here](https://github.com/robolectric/robolectric/issues/5104)
+
+!!! Disclaimer
+
+      It is important to note that UiAutomator tests cannot run as shared tests. 
+      That's because Google has not provided support for it yet, and might never do it.
+      Only Compose tests via ComposeTestRules as well as Espreso tests are supported
+
 
 ## Conclusion
```

Comment:
Conclusions would need to be rewritten I believe... but I do not know still what to write here since I changed 80% of the original doc